pkg/image/registry/imagestreamimport/rest.go

```diff
@@ -78,10 +86,52 @@ func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, err
 
 	inputMeta := isi.ObjectMeta
 
-	if err := rest.BeforeCreate(Strategy, ctx, obj); err != nil {
+	if err := rest.BeforeCreate(r.strategy, ctx, obj); err != nil {
 		return nil, err
 	}
 
+	// Check if the user is allowed to create Images or ImageStreamMappings.
+	// In case the user is allowed to create them, do not validate the ImageStreamImport
+	// registry location against the registry whitelist, but instead allow to create any
+	// image from any registry.
+	user, ok := kapi.UserFrom(ctx)
+	if !ok {
+		return nil, kapierrors.NewBadRequest("unable to get user from context")
+	}
+	isCreateImage, err := r.sarClient.Create(
+		&authorizationapi.SubjectAccessReview{
+			User: user.GetName(),
+			Action: authorizationapi.Action{
+				Verb:     "create",
+				Group:    api.GroupName,
+				Resource: "images",
+			},
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	isCreateImageStreamMapping, err := r.sarClient.Create(
+		&authorizationapi.SubjectAccessReview{
+			User: user.GetName(),
+			Action: authorizationapi.Action{
+				Verb:     "create",
+				Group:    api.GroupName,
+				Resource: "imagestreammapping",
+			},
+		},
+	)
+	if err != nil {
```

Comment:
Isn't image creation enough? I thought the check should go like `canCreateIS || canCreateISM`?